continuous-integration/check.sh

```diff
@@ -0,0 +1,248 @@
+    if [[ $BASE == "ubuntu" ]]; then
+        jhbuild sysdeps --install $1
+    fi
+    jhbuild build $(jhbuild list $1 | sed '$d')
+}
+
+function do_Save_Files(){
+    echo
+    echo '-- Saving build files --'
+    mkdir -p /cwd/SAVED/$OS
+
+    cp -r ~/jhbuild /cwd/SAVED/$OS/jhbuild
+    cp -r ~/.local  /cwd/SAVED/$OS/.local
+    echo '-- Done --'
+}
+
+function do_Get_Files(){
+    echo
+    echo '-- Restoring build files --'
+
+    cp -r /cwd/SAVED/$OS/jhbuild ~/jhbuild
+    cp -r /cwd/SAVED/$OS/.local  ~/.local
+    echo '-- Done --'
+}
+
+function do_Clean_Image(){
+    echo
+    echo '-- Cleaning image --'
+
+    # Clean the environment
+    rm -f /root/jhbuild/install/lib/libmozjs-31.a
+    jhbuild clean
+
+    if [[ $BASE == "ubuntu" ]]; then
+        # Base dependencies
+        apt-get -y -qq remove --purge build-essential git clang patch
+        apt-get -y -qq remove --purge autotools-dev autoconf gettext pkgconf autopoint yelp-tools
+        apt-get -y -qq remove --purge docbook docbook-xsl libtext-csv-perl
+        apt-get -y -qq remove --purge zlib1g-dev
+        apt-get -y -qq remove --purge libtool libicu-dev libnspr4-dev
+
+        # JHBuild dependencies management
+        apt-get -y -qq remove --purge policykit-1
+        apt-get -y -qq remove --purge apt-file
+
+        apt-get -y autoremove
+        apt-get -y clean
+        rm -rf /var/lib/apt/lists/*
+
+    elif [[ $BASE == "fedora" ]]; then
+        # Base dependencies
+        dnf -y remove @c-development @development-tools redhat-rpm-config gnome-common
+        dnf -y remove pygobject2 dbus-python perl-Text-CSV perl-XML-Parser gettext-devel gtk-doc ninja-build
+        dnf -y remove zlib-devel libffi-devel
+        dnf -y remove libtool libicu-devel nspr-devel
+        dnf -y remove which || true
+
+        dnf -y autoremove
+        dnf -y clean all
+    fi
+    echo
+    echo '-- Image done --'
+}
+
+# ----------- GJS -----------
+cd /cwd
+
+# Show some environment info
+echo
+echo '-- Environment --'
+echo "Running on: $BASE $OS"
+echo "Doing: $1"
+
+if [[ $1 == "BUILD_MOZ" ]]; then
+    do_Install_Base_Dependencies
+
+    # Compiler to be used
+    echo
+    echo '-- Compiler in use --'
+    $CC --version
+
+    do_Patch_JHBuild
+    do_Build_JHBuild
+    do_Build_Mozilla
+    do_Save_Files
+
+elif [[ $1 == "SAVE_MOZ" ]]; then
+    do_Get_Files
+
+elif [[ $1 == "GJS" ]]; then
+    do_Install_Base_Dependencies
+    do_Install_Dependencies
+
+    # Compiler to be used
+    echo
+    echo '-- Compiler in use --'
+    $CC --version
+
+    do_Patch_JHBuild
+    do_Build_JHBuild
+    do_Build_Package_Dependencies gjs
+
+    # Build the latest commit (merged or from a PR) of Javascript Bindings for GNOME
+    echo
+    echo '-- gjs build --'
+    cd ../gjs
+    git log --pretty=format:"%h %cd %s" -1
+    echo
+    jhbuild run ./autogen.sh --prefix /root/jhbuild/install --enable-compile-warnings=error --enable-installed-tests --with-xvfb-tests
```

Comment:
These three lines (and `jhbuild run make check` below) can all be condensed into `jhbuild make --check`, which will configure, make, make check, and make install whatever the current commit in the repository is. To get the configure options, it would be best to create a `~/.config/jhbuildrc` in the setup phase, with contents such as this:
```py
module_autogenargs['gjs'] = '--enable-compile-warnings=error --enable-installed-tests --with-xvfb-tests'
module_makeargs['gjs'] = '-sj2'
```